src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration.Tests/MockHttpClientTransport.cs

```diff
@@ -204,43 +295,72 @@ void VerifyRequest(HttpRequestMessage request)
+        public bool TrySerializeConfigurationSetting(ConfigurationSetting setting, byte[] buffer, out int written)
         {
-            if (responseContent == null) return null;
-
-            StringBuilder requestContent = new StringBuilder();
-            requestContent.AppendFormat("{{\"key\":\"{0}\",", responseContent.Key);
-            requestContent.AppendFormat("\"label\":\"{0}\",", responseContent.Label);
-            requestContent.AppendFormat("\"value\":\"{0}\",", responseContent.Value);
-            requestContent.AppendFormat("\"content_type\":\"{0}\",", responseContent.ContentType);
-            requestContent.AppendFormat("\"etag\":\"{0}\",", responseContent.ETag.ToString());
-            requestContent.AppendFormat("\"last_modified\":\"{0}\",", DateTimeOffset.Now.UtcDateTime.ToString("yyyy'-'MM'-'dd'T'HH':'mm':'ssK"));
-            if (locked.HasValue) requestContent.AppendFormat("\"locked\":{0},", locked);
-            requestContent.AppendFormat("\"tags\":{{");
+            try
+            {
+                var writer = new FixedSizedBufferWriter(buffer);
+                Utf8JsonWriter json = new Utf8JsonWriter(writer);
+                SerializeSetting(ref json, setting);
+                json.Flush();
+                written = (int)json.BytesWritten;
+                return true;
+            }
+            catch (ArgumentException)
+            {
+                written = 0;
+                return false;
+            }
+        }
 
-            bool first = true;
-            foreach (var tag in responseContent.Tags)
+        protected void SerializeSetting(ref Utf8JsonWriter json, ConfigurationSetting setting)
+        {
+            json.WriteStartObject();
+            json.WriteString("key", setting.Key);
+            json.WriteString("label", setting.Label);
+            json.WriteString("value", setting.Value);
+            json.WriteString("content_type", setting.ContentType);
+            if (setting.Tags != null)
             {
-                if (first)
+                json.WriteStartObject("tags");
+                foreach (var tag in setting.Tags)
                 {
-                    requestContent.AppendFormat("\"{0}\":\"{1}\"", tag.Key, tag.Value);
-                    first = false;
+                    json.WriteString(tag.Key, tag.Value);
                 }
-                else
+                json.WriteEndObject();
+            }
+            if (setting.ETag != default) json.WriteString("etag", setting.ETag.ToString());
+            if (setting.LastModified.HasValue) json.WriteString("last_modified", setting.LastModified.Value.ToString());
+            if (setting.Locked.HasValue) json.WriteBoolean("locked", setting.Locked.Value);
+            json.WriteEndObject();
+        }
+
+        protected ReadOnlyMemory<byte> Serialize(ConfigurationSetting setting)
+        {
+            if (setting == null) return null;
+
+            ReadOnlyMemory<byte> content = default;
```

Comment:
We have way too much non-trivial code duplicated. Can you add an IBufferWriter implementation that List<byte> internally to avoid having Try/expand loops duplicated?

Reply:
We talked offline about this and decided to leave it as it is. 
There is a new update on Microsoft.Bcl.Json.Sources that includes an ArrayBufferWriter. Once it is available in NuGet, we will consume it